### PR TITLE
[TOPI] deformable_conv2d in NHWC

### PR DESCRIPTION
**Pull request by vinx13**
@comaniac

### CODE REVIEW

File: python/tvm/topi/generic/nn.py [hunk 1]
```diff
@@ -462,6 +462,23 @@ def schedule_deformable_conv2d_nchw(outs):
     return _default_schedule(outs, False)
 
 
+def schedule_deformable_conv2d_nhwc(outs):
+    """Schedule for deformable_conv2d_nhwc
```
Comment (comaniac) [inline python/tvm/topi/generic/nn.py:466]:
```suggestion
    """Schedule for deformable_conv2d_nhwc.
    We only use the default schedule here and rely on auto_scheduler.
```

File: python/tvm/topi/testing/deformable_conv2d_python.py [hunk 2]
```diff
@@ -119,3 +119,51 @@ def _bilinear(n, c, h, w):
         b_np[n, f, h, w] += np.tensordot(a_deform[n, c, h, w], w_np[f, c])
 
     return b_np
+
+
+def deformable_conv2d_nhwc_python(
+    a_np, offset_np, w_np, stride, padding, dilation, deformable_groups, groups
+):
+    """Deformable convolution operator in NCHW layout.
```
Comment (comaniac) [inline python/tvm/topi/testing/deformable_conv2d_python.py:127]:
```suggestion
    """Deformable convolution operator in NHWC layout.
```

File: tests/python/topi/python/test_topi_deformable_conv2d.py [hunk 3]
```diff
@@ -112,11 +115,94 @@ def check_device(device):
...
+            a = tvm.nd.array(a_np, ctx)
+            offset = tvm.nd.array(offset_np, ctx)
+            w = tvm.nd.array(w_np, ctx)
+            c = tvm.nd.empty(c_np.shape, dtype=c_np.dtype, ctx=ctx)
+
+            func = tvm.build(s, [A, Offset, W, C], device)
+            func(a, offset, w, c)
+            tvm.testing.assert_allclose(c.asnumpy(), c_np, rtol=1e-5)
+
+    for device in ["llvm"]:
```
Comment (comaniac) [inline tests/python/topi/python/test_topi_deformable_conv2d.py:194]: Could we also check CUDA?

**Review by comaniac — CHANGES_REQUESTED** [review]
Thanks for the PR! Just a few comments.

File: tests/python/topi/python/test_topi_deformable_conv2d.py (hunk 3 above)
Comment (merrymercy) [reply tests/python/topi/python/test_topi_deformable_conv2d.py:194]: I think we cannot because we don't have a valid schedule

Comment (comaniac) [reply tests/python/topi/python/test_topi_deformable_conv2d.py:194]: Ah nvm. I forgot the default schedule doesn't work for GPU.

**Review by comaniac — APPROVED** [review]
LGTM

**Comment by comaniac** [comment]
Thanks @vinx13 @merrymercy

**Pull request merged**